### PR TITLE
feat: add parking lot compatable api

### PR DESCRIPTION
**Pull request by driftluo**
Introduce `parking_lot` to tentacle

### CODE REVIEW

File: tentacle/src/channel/bound.rs [hunk 1]
```diff
@@ -47,7 +48,10 @@ pub fn channel<T>(buffer: usize) -> (Sender<T>, Receiver<T>) {
 
     let tx = BoundedSenderInner {
         inner: inner.clone(),
+        #[cfg(not(feature = "parking_lot"))]
         sender_task: Arc::new(Mutex::new(SenderTask::new())),
+        #[cfg(feature = "parking_lot")]
+        sender_task: Arc::new(crate::lock::const_mutex(SenderTask::new())),
```
Comment (quake) [inline tentacle/src/channel/bound.rs:54]: I think there are no different than a normal `new` fn:
```suggestion
sender_task: Arc::new(Mutex::new(SenderTask::new()))
```

File: tentacle/src/lock/mod.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#![allow(dead_code)]
```
Comment (quake) [inline tentacle/src/lock/mod.rs:1]: unnecessary hint?

File: tentacle/Cargo.toml [hunk 3]
```diff
@@ -25,6 +25,8 @@ tokio-util = { version = "0.6.0", features = ["codec"] }
 log = "0.4"
 bytes = "1.0.0"
 thiserror = "1.0"
+once_cell = "1.0"
+parking_lot = { version = "0.11", optional = true }
```
Comment (quake) [inline tentacle/Cargo.toml:29]: need to add a CI task to enable this feature test and clippy

**Review by quake — CHANGES_REQUESTED** [review]

File: tentacle/src/lock/mod.rs (hunk 2 above)
Comment (driftluo) [reply tentacle/src/lock/mod.rs:1]: RwLock are not used

File: tentacle/Cargo.toml (hunk 3 above)
Comment (driftluo) [reply tentacle/Cargo.toml:29]: add a feature check ci，since the interface is consistent with std, clippy check is not added

**Review by quake — APPROVED** [review]

**Review by TheWaWaR — APPROVED** [review]

**Pull request merged**